obspy/dyna/core.py

```diff
@@ -0,0 +1,827 @@
+        stats.dyna['LOCATION'] = stats['location']
+        stats.dyna['STREAM'] = stats['channel']
+
+        stats.dyna['NDATA'] = stats['npts']
+        stats.dyna['SAMPLING_INTERVAL_S'] = stats['delta']
+
+        # conditional: seconds or milliseconds
+        # = stats.starttime
+        stats.dyna['DATE_TIME_FIRST_SAMPLE_YYYYMMDD_HHMMSS'] \
+                = stats['starttime']
+
+        # Warining?
+        if stats['calib'] != 1.0:
+            pass
+
+        # data type qis acceleration, acc, SA
+        # DATA_TYPE == "ACCELERATION"
+        # DATA_TYPE == "ACCELERATION RESPONSE SPECTRUM"
+        if is_acceleration:
+            # key_peak = "PGA_CM/S^2", "PGA_CM_S_2",
+            # key_peak_time = "TIME_PGA_S"
+            pass
+
+        # data type is velocity, vel, PSV
+        # DATA_TYPE == "VELOCITY"
+        # DATA_TYPE == "PSEUDO-VELOCITY RESPONSE SPECTRUM"
+        if is_velocity:
+            # key_peak = "PGV_CM/S", "PGV_CM_S",
+            # key_peak_time = "TIME_PGV_S"
+            pass
+
+        # data type is displacement, dis, SD
+        # DATA_TYPE == "DISPLACEMENT", not written?
+        # DATA_TYPE == "DISPLACEMENT RESPONSE SPECTRUM"
+        if is_displacement:
+            # key_peak = "PGD_CM"
+            # key_peak_time = "TIME_PGD_S"
+            pass
+
+        return
+
+
+def isDYNA(filename):
+    """
+    Checks whether a file is a DYNA 1.0 ASCII file or not.
+
+    :type filename: str
+    :param filename: Name of the DYNA file to be checked.
+    :rtype: bool
+    :return: ``True`` if a DYNA 1.0 ASCII file.
+
+    .. rubric:: Example
+
+    >>> isDYNA("/path/to/IT.ARL..HGE.D.20140120.071240.X.ACC.ASC")  #doctest: +SKIP  # NOQA
+    True
+    """
+    # first eleven chars should contain 'EVENT_NAME:'
+    try:
+        temp = open(filename, 'rt').read(11)
+    except:
+        return False
+    if temp != 'EVENT_NAME:':
+        return False
+    f = open(filename, 'rt')
+    for i in range(55):
+        line = f.readline()
+        if i == 47 and line[:23] == 'HEADER_FORMAT: DYNA 1.0':
+            return True
+    return False
+
+
+def isITACA(filename):
+    """
+    Checks whether a file is a ITACA ASCII file or not.
+
+    :type filename: str
+    :param filename: Name of the ITACA file to be checked.
+    :rtype: bool
+    :return: ``True`` if a ITACA ASCII file.
+
+    .. rubric:: Example
+
+    >>> isITACA("/path/to/19971014_152309ITDPC_NCR__WEX.DAT")  #doctest: +SKIP
+    True
+    """
+    # first eleven chars should contain 'EVENT_NAME:'
+    try:
+        temp = open(filename, 'rt').read(11)
+    except:
+        return False
+    if temp != 'EVENT_NAME:':
+        return False
+    # first 12 chars of line 8 should contain 'MAGNITUDE_S:'
+    f = open(filename, 'rt')
+    for i in range(8):
+        line = f.readline()
+        if i == 7 and line[:12] != 'MAGNITUDE_S:':
+            return False
+    # filename must match the following regexp:
+    # ^\d{8}_\d{6}\w{5}_\w{5}\w{3}.\w{3}$
```

Comment:
Keep for the moment, as we could extract the filename only part (remove path).
But probably filename checking should be removed to support file-like objects as well.